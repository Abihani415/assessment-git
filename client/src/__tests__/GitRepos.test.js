import React from "react";
import { screen, render, fireEvent, act, waitForElementToBeRemoved } from '../test-utils'
import "@testing-library/jest-dom";
import GitRepo from "../component/gitRepo";


describe("Take GitRepo Component snapshot" , () => {
  it("Git Repo List", async () => {
    render(<GitRepo />)
    const inputField = screen.getByTestId('handle-test-id');
    const submitBtn = screen.getByTestId('submit-test-id');

    fireEvent.change(inputField, {target: {value: 'apache'}});
    await act(async () => {
        fireEvent.click(submitBtn);
    });
    await waitForElementToBeRemoved(() => screen.queryAllByTestId("loading-test-id"))
    expect(screen.getByTestId("repo-wrapper-test-id")).toMatchSnapshot();
  })
})