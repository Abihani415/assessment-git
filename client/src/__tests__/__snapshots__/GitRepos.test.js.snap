// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take GitRepo Component snapshot Git Repo List 1`] = `
<div
  data-testid="repo-wrapper-test-id"
>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          tapestry3
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          4
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Tapestry 3
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/tapestry3"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          apr-iconv
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          16
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Portable Runtime iconv
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/apr-iconv"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          tapestry4
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          6
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Tapestry 4
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/tapestry4"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          sling-old-svn-mirror
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          222
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Sling
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/sling-old-svn-mirror"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          xalan-j
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          23
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Xalan Java
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/xalan-j"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          etch
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          17
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Etch
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/etch"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          apr
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          348
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache Portable Runtime
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/apr"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          stdcxx
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          53
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Mirror of Apache C++ Standard Library
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/stdcxx"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          zookeeper
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          10095
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Apache ZooKeeper
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/zookeeper"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div
    class="repo-wrapper card my-3 pb-3"
  >
    <div
      class="d-flex align-items-start justify-content-between p-3 border-bottom"
    >
      <div>
        <span
          class="card-heading"
        >
          Name:
        </span>
          
        <span
          class="reponame "
        >
          lucenenet
        </span>
      </div>
      <div>
        <i
          class="fa fa-star"
        />
        <span
          class="text-dark"
        >
          1713
        </span>
      </div>
    </div>
    <div
      class="repo_desc p-3"
    >
      <span
        class="card-heading"
      >
        Description:  
      </span>
      <span
        class="description"
      >
        Apache Lucene.NET
      </span>
    </div>
    <div
      class="d-flex align-items-start justify-content-between px-3"
    >
      <div
        class="reponame"
      >
        <span
          class="card-heading"
        >
          Owner Name:  
        </span>
        <span>
          apache
        </span>
      </div>
      <div
        class="reponame"
      >
        <a
          href="https://github.com/apache/lucenenet"
          rel="noreferrer"
          target="_blank"
        >
          Visit Repo
        </a>
      </div>
    </div>
  </div>
  <div>
    <span>
      Total Repos : 
    </span>
    <ul
      class="pagination"
    >
      <li
        class="page-item"
      >
        <a
          class="page-link"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
          >
            ‹
          </span>
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          role="button"
          tabindex="0"
        >
          NaN
          NaN
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <span
          class="page-link"
          disabled=""
        >
          <span
            aria-hidden="true"
          >
            ›
          </span>
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
